### DISCUSSION TITLE
CRC mismatch

### DISCUSSION THREAD

**Discussion by mcschuetz** [icsharpcode/SharpZipLib#773] [Q&A]
Hello all,

I always get this error with exactly one specific file:

ICSharpCode.SharpZipLib.Zip.ZipException: CRC mismatch
   at ICSharpCode.SharpZipLib.Zip.ZipInputStream.CompleteCloseEntry(Boolean testCrc)
   at ICSharpCode.SharpZipLib.Zip.ZipInputStream.BodyRead(Byte[] buffer, Int32 offset, Int32 count)

Maybe an important information, the zip is loaded from the BlobStorage could it be related to that?

I just don't understand why with 600 other zip files in the system everything works.

Unfortunately I can't reveal the zip file for privacy reasons. 

Has anyone perhaps struggled with this before and knows a reason/solution?

Thanks a lot 

Best regards

### Version of SharpZipLib
1.3.3

**Comment by piksel** [comment]
The zip file is probably corrupted. The CRC is used to confirm that the contents of the extracted file matches the contents of the input file, and for some reason it doesn't for your file.
If you are lucky, the only corrupted part is the actual CRC and the contents are still fine. In that case you can try decompressing it with another tool like 7z (which I think can ignore CRC checks), and then re-compressing it again (if that is your goal).
